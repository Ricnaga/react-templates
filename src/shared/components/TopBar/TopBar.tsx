import { Grid, Link } from '@chakra-ui/react';

import { HOME, TABLE_SAMPLE, TODO } from '../../../application/routes/paths';

const links = [
  {
    href: HOME,
    children: 'Home',
  },
  {
    href: TODO,
    children: 'TODO',
  },
  {
    href: TABLE_SAMPLE,
    children: 'Table Sample',
  },
];

export function TopBar() {
  return (
    <Grid my={4}>
      <nav>
        {links.map(({ href, children }) => (
          <Link key={children} href={href} mr={2} fontSize="lg">
            {children}
          </Link>
        ))}
      </nav>
    </Grid>
  );
}
