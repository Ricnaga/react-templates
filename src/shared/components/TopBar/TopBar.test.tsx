import { renderWithTheme } from '@application/test/testing-library';

import { TopBar } from './TopBar';

let containerElement: HTMLElement;

describe('Component: TopBar', () => {
  beforeEach(() => {
    const { container } = renderWithTheme(<TopBar />);
    containerElement = container;
  });

  it('should render correctly', () => {
    expect(containerElement).toMatchSnapshot();
  });
});
