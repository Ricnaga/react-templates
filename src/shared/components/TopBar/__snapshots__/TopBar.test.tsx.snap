// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TopBar should render correctly 1`] = `
<div>
  <div
    class="css-1fbc7bo"
  >
    <nav>
      <a
        class="chakra-link css-1ehl8a5"
        href="/"
      >
        Home
      </a>
      <a
        class="chakra-link css-1ehl8a5"
        href="/todo"
      >
        TODO
      </a>
      <a
        class="chakra-link css-1ehl8a5"
        href="/table"
      >
        Table Sample
      </a>
    </nav>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
